### DISCUSSION TITLE
Error connecting to database

### DISCUSSION THREAD

**Discussion by jsooter** [phalcon/cphalcon#15681] [Q&A]
Using version 4.0, when I try to connect to the database I get thefollowing error
`SQLSTATE[08006] [7] could not connect to server: No such file or directory Is the server running locally and accepting connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?`

This is how I'm setting up the connection
```PHP
$container->set('session', function() use ($config){
    // Create a connection
    $connection = new \Phalcon\Db\Adapter\Pdo\Postgresql((array)$config->session_db); // <-- error is here

    $session = New \Phalcon\Session\Adapter\SessionDatabase($connection,'session_data');

    $session->start();
    return $session;
});
```
The config looks like this
```PHP
    'session_db' => array(
        'host'     => 'db',
        'username' => 'xxxxx',
        'password' => 'xxxxx',
        'dbname'   => 'session',
        'schema'   => 'public'
    ),
```

The database is running in another Docker container called `db`. It looks like Phalcon expects a local socket connection for some reason.

Anyone know why it's looking for a local socket connection?

**Comment by quasipickle** [comment]
My guess is it's looking for a socket because you haven't specified a hostname.  I'm not too familiar with Docker, but I don't think specifying `db` is sufficient.

Maybe try specifying `localhost` and the correct port.

**Comment by jsooter** [comment]
Thanks @quasipickle. I finally figured out the issue. For some reason the config array was loading as a string.

**Comment by GammaGames** [comment]
👋 It's something with the config's properties. Using `new \Phalcon\Config(['database' => ['host' => 'db', 'port' => '5432']])`:

1. `var_dump($config->database)` shows protected properties
    ```php
    object(Phalcon\Config)#25 (4) {
      ["data":protected]=>array(2) {
        ["host"]=>string(2) "db"
        ["port"]=>string(4) "5432"
      }
      ["insensitive":protected]=>bool(true)
      ["lowerKeys":protected]=>array(2) {
        ["host"]=>string(4) "host"
        ["port"]=>string(4) "port"
      }
      ["pathDelimiter":protected]=>NULL
    }
    ```

2. `var_dump((array)$config->database)` prints them out with *protected:
    ```php
    array(4) {
      ["*data"]=>array(2) {
        ["host"]=>string(2) "db"
        ["port"]=>string(4) "5432"
      }
      ["*insensitive"]=>bool(true)
      ["*lowerKeys"]=>array(2) {
        ["host"]=>string(4) "host"
        ["port"]=>string(4) "port"
      }
      ["*pathDelimiter"]=>NULL
    }
    ```
3. `var_dump($config->database->toArray())` converts correctly and works as expected:
    ```php
    array(2) {
      ["host"]=>string(2) "db"
      ["port"]=>string(4) "5432"
    }
    ```
